Implement revoke admin cli support

joaosa opened the pull request:
We couldn't really remove admins from the cli, so this should fix that. I wanted to cleanup the code a little bit, but I figured it would be more important to ship this rather than adjust existing functionality for code reuse.

- Added a wrapper to use the contract's `revoke_admin` method;
- Added the clap cli boilerplate to use it.

joaosa committed:
Wrap contract revoke_admin in a helper function

joaosa committed:
Add revoke_admin cli support

joaosa committed:
Add missing revoke_admin import

joaosa commented on cli/src/utils.rs:
```diff
@@ -1011,6 +1011,35 @@ pub async fn grant_admin<S: ::ethers::providers::Middleware + 'static>(
+
+    info!("estimated grant gas cost {:#?}", claim_tx.tx.gas().unwrap());
+
+    send_tx(&claim_tx.tx, client, retries).await?;
```
Yes. The `?` operator is going to propagate any error back to the caller function because this function has a `Result<(), Box<dyn std::error::Error>>` return type.

That's a great suggestion. It would definitely help in terms of usability. I'll look into it :)

gammazero reviewed (commented):
Other than logging already suggested, this looks good. My rust knowledge is minimal for now, so I might be missing things.

Should there be a test for granting and revoking admin?

joaosa commented:
> Other than logging already suggested, this looks good. My rust knowledge is minimal for now, so I might be missing things.
> 
> Should there be a test for granting and revoking admin?

Thank you! I agree we should have this. I will have to learn how to use the test tooling to pull it off (which is probably a good thing anyway). Should it be a necessary to merge the PR?

joaosa committed:
Get feedback transaction feedback when adding/revoking admins

joaosa committed:
Remove unused test import

joaosa committed:
Add grant admin integration test

joaosa committed:
Add revoke admin integration test

joaosa commented:
@travis @gammazero 

- added some logging so users can get the message address for the grant/revoke operations they ran and check them on GLIF right away;
- added integration tests and verified things are working fine

I should also note that as things are, it's not possible to run integration tests locally as we're missing three secrets and we have to rely on CI. However, it's possible to troubleshoot what's going on with all these invocations on calibrationnet. [Here's](https://calibration.filfox.info/en/address/0x037c8683a23f9085c87fd15489a90f7fec5f18d2?t=4) an example of the last CI run with all the integration test transactions.

joaosa committed:
Fix typo to get the transaction hash

gammazero reviewed (approved):
LGTM - Thanks for adding the admin grant/revoke tests. Since these pass I think we can call this good
